packages/core/src/Stores/client-store.js

```diff
@@ -2240,5 +2241,25 @@ export default class ClientStore extends BaseStore {
             resolve(get_financial_assessment);
         });
     }
+
+    @action.bound
+    setTwoFAStatus(status) {
+        this.has_enabled_two_fa = status;
+    }
+
+    @action.bound
+    getTwoFAStatus() {
+        return new Promise(resolve => {
+            WS.authorized.accountSecurity({ account_security: 1, totp_action: 'status' }).then(response => {
+                if (response.error) {
+                    resolve(response.error);
+                } else {
+                    const is_enabled = !!getPropertyValue(response, ['account_security', 'totp', 'is_enabled']);
+                    this.setTwoFAStatus(is_enabled);
+                    resolve(response);
+                }
+            });
```

Comment:
This part of the code already exist in the component, is it needed in both places?

Reply:
I have changed this and updated PR